MSAL/test/integration/native_auth/end_to_end/mfa/MSALNativeAuthSignInWithMFAEndToEndTests.swift

```diff
@@ -0,0 +1,226 @@
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.  
+
+import Foundation
+import XCTest
+import MSAL
+
+final class MSALNativeAuthSignInWithMFAEndToEndTests: MSALNativeAuthEndToEndPasswordTestCase {
+
+    func test_signInUsingPasswordWithMFASubmitWrongChallengeResendChallengeThen_completeSuccessfully() async throws {
+        throw XCTSkip("Missing username for MFA user")
+        guard let username = retrieveUsernameForSignInUsernamePasswordAndMFA(), 
+                let password = await retrievePasswordForSignInUsername(),
+                let awaitingMFAState = await signInUsernameAndPassword(username: username, password: password)
+        else {
+            XCTFail("Something went wrong")
+            return
+        }
+        
+        // Request to send challenge to the default strong auth method
+        let mfaExpectation = expectation(description: "mfa")
+        let mfaDelegateSpy = MFARequestChallengeDelegateSpy(expectation: mfaExpectation)
+        
+        awaitingMFAState.requestChallenge(delegate: mfaDelegateSpy)
+        
+        await fulfillment(of: [mfaExpectation])
+        
+        guard mfaDelegateSpy.onVerificationRequiredCalled, let mfaRequiredState = mfaDelegateSpy.newStateMFARequired else {
+            XCTFail("Challenge not sent to MFA method")
+            return
+        }
+        
+        // Now submit the wrong email OTP code
+        let submitWrongChallengeExpectation = expectation(description: "submitChallenge")
+        let mfaSubmitWrongChallengeDelegateSpy = MFASubmitChallengeDelegateSpy(expectation: submitWrongChallengeExpectation)
+
+        mfaRequiredState.submitChallenge(challenge: "000000", delegate: mfaSubmitWrongChallengeDelegateSpy)
```

Comment:
nit: we take the real OTP and add a symbol to it, so we're 100% sure that it's invalid. There's a 0.00000% change that "00000" is the actual value that's sent :D

Reply:
You're right, there is a small chance that this code is valid. But retrieving the real code is slower than hardcode a value. Given that this specific E2E test is already slow, it's ok to don't be 100% sure in this case :)

Reply:
is it possible to send a string like `"wrong code"`, for example?

Reply:
Let me check with the ESTS team

Reply:
There should not be any issue because the code is treated as string. So, I used the string "wrong_code". Thanks Diego